src/main/java/com/nekomaster1000/infernalexp/entities/GlowsilkMothEntity.java

```diff
@@ -0,0 +1,135 @@
+import net.minecraft.util.math.MathHelper;
+import net.minecraft.util.math.vector.Vector3d;
+import net.minecraft.world.World;
+
+import javax.annotation.Nullable;
+
+public class GlowsilkMothEntity extends AmbientEntity {
+    private BlockPos spawnPosition;
+
+    public GlowsilkMothEntity(EntityType<? extends GlowsilkMothEntity> type, World worldIn) {
+        super(type, worldIn);
+    }
+
+    //ATTRIBUTES
+    public static AttributeModifierMap.MutableAttribute setCustomAttributes() {
+        return MobEntity.func_233666_p_()
+                .createMutableAttribute(Attributes.MAX_HEALTH, 6.0D);
+    }
+
+    /**
+     * Returns the volume for the sounds this mob makes.
+     */
+    protected float getSoundVolume() {
+        return 0.1F;
+    }
+
+    /**
+     * Gets the pitch of living sounds in living entities.
+     */
+    protected float getSoundPitch() {
+        return super.getSoundPitch() * 0.95F;
+    }
+
+    @Nullable
+    public SoundEvent getAmbientSound() {
+        return SoundEvents.ENTITY_BAT_AMBIENT;
+    }
+
+    protected SoundEvent getHurtSound(DamageSource damageSourceIn) {
+        return SoundEvents.ENTITY_BAT_HURT;
+    }
+
+    protected SoundEvent getDeathSound() {
+        return SoundEvents.ENTITY_BAT_DEATH;
+    }
+
+    /**
+     * Returns true if this entity should push and be pushed by other entities when colliding.
+     */
+    public boolean canBePushed() {
+        return false;
+    }
+
+    protected void collideWithEntity(Entity entityIn) {
+    }
+
+    protected void collideWithNearbyEntities() {
+    }
+
+    /**
+     * Called to update the entity's position/logic.
+     */
+    public void tick() {
+        super.tick();
+        this.setMotion(this.getMotion().mul(1.0D, 1.0D, 1.0D));
+    }
+
+    protected void updateAITasks() {
+        super.updateAITasks();
+        if (this.spawnPosition != null && (!this.world.isAirBlock(this.spawnPosition) || this.spawnPosition.getY() < 1)) {
+                this.spawnPosition = null;
+        }
+
+        if (this.spawnPosition == null || this.rand.nextInt(30) == 0 || this.spawnPosition.withinDistance(this.getPositionVec(), 2.0D)) {
+                this.spawnPosition = new BlockPos(this.getPosX() + (double)this.rand.nextInt(7) - (double)this.rand.nextInt(7), this.getPosY() + (double)this.rand.nextInt(6) - (double)this.rand.nextInt(2), this.getPosZ() + (double)this.rand.nextInt(7) - (double)this.rand.nextInt(7));
+        }
+
+            double d2 = (double)this.spawnPosition.getX() + 0.5D - this.getPosX();
+            double d0 = (double)this.spawnPosition.getY() + 0.1D - this.getPosY();
+            double d1 = (double)this.spawnPosition.getZ() + 0.5D - this.getPosZ();
+            Vector3d vector3d = this.getMotion();
+            Vector3d vector3d1 = vector3d.add((Math.signum(d2) * 0.5D - vector3d.x) * (double)0.3F, (Math.signum(d0) * 0.7F - vector3d.y) * (double)0.3F, (Math.signum(d1) * 0.5D - vector3d.z) * (double)0.3F);
```

Comment:
Why do you use floats and immediately typecast them to doubles?
```suggestion
            Vector3d vector3d1 = vector3d.add((Math.signum(d2) * 0.5D - vector3d.x) * 0.3D, (Math.signum(d0) * 0.7F - vector3d.y) * 0.3D, (Math.signum(d1) * 0.5D - vector3d.z) * 0.3D);
```